src/routes.rs

```diff
@@ -120,6 +120,22 @@ impl IntoResponse for DispenseInfoResponse {
     }
 }
 
+async fn has_reached_dispense_limit(
+    dispense_tracker: &SharedDispenseTracker,
+    address: Address,
+    interval: u64,
+) -> bool {
+    let mut dispense_tracker = dispense_tracker.lock().await;
```

Comment:
You don't need to use `tokio::Mutex` because it uses normal `Mutex` under the hood. You can use normal mutex too because you lock it for small amount of time.